AgentDiscoveries-Frontend/app/src/components/submit-forms/agent-form.js

```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import {
+    Form,
+    FormGroup,
+    FormControl,
+    Button,
+    ControlLabel
+} from "react-bootstrap";
+import { handleSubmit } from "./submit-utilities"
+import { Message } from "../message"
+
+export class CreateAgent extends React.Component {
+
+    constructor(props) {
+        super();
+        this.state = {
+            "submitForm": props.submitForm,
+            "onSubmit": props.onSubmit
+        }
+    }
+
+    render(props) {
+        return (
+            <div className="col-md-12">
+                <Form onSubmit={this.state.onSubmit}>
+                    <h3>Create Agent</h3>
+
+                    <FormGroup>
+                        <ControlLabel>First Name</ControlLabel>
+                        <FormControl type="text" required
+                            inputRef={firstName => this.state.submitForm.firstName = firstName}

```

Comment:
I'm pretty sure you should never modify state directly in react. See https://reactjs.org/docs/forms.html